lib/src/rules/noop_primitive_operations.dart

```diff
@@ -66,10 +69,27 @@ class _Visitor extends SimpleAstVisitor<void> {
     }
   }
 
+  @override
+  void visitInterpolationExpression(InterpolationExpression node) {
+    var expression = node.expression;
+    if (expression is MethodInvocation &&
+        expression.methodName.name == 'toString' &&
+        expression.argumentList.arguments.isEmpty) {
+      rule.reportLint(expression.methodName);
+    }
+  }
+
   @override
   void visitMethodInvocation(MethodInvocation node) {
     var type = node.realTarget?.staticType;
-    if (type == null) return;
+    if (type == null) {
+      // print(xxx.toString())
+      if (node.methodName.name == 'print' &&
+          node.methodName.staticElement?.library?.name == 'dart.core') {
```

Comment:
I wonder if we shouldn't prefer:

```dart
node.methodName.staticElement?.library?.isDartCore ?? false
```

(and if so, update `AvoidPrint` similarly).

Dunno.

@scheglov: do you all have comventions in analyzer?

Reply:
`isDartCore` is definitely better.

Also, I would have added an extension like `isDartCorePrint` on `Element?` and checked that it is `FunctionElement` with the required name and library.

Reply:
Playing around with that now.  Stay tuned! 😄

Reply:
Landed.  Thanks @scheglov!

@a14n: could you migrate to using `isDartCorePrint` as introduced by https://github.com/dart-lang/linter/pull/3382?